controllers/admin/AdminImportController.php

```diff
@@ -3762,44 +3797,66 @@ public function storeContactImportOne($info, $shop_is_feature_active, $regenerat
+                    if ($fieldsValidationResult !== true || isset($langFieldsValidationResult) && $langFieldsValidationResult !== true) {
+                        $this->errors[] = ($fieldsValidationResult !== true ? $fieldsValidationResult : '').(isset($langFieldsValidationResult) && $langFieldsValidationResult !== true ? $langFieldsValidationResult : '').
                             Db::getInstance()->getMsgError();
                     }
                 }
             }
         }
 
-        $res = false;
-        if (($field_error = $store->validateFields(UNFRIENDLY_ERROR, true)) === true &&
-            ($lang_field_error = $store->validateFieldsLang(UNFRIENDLY_ERROR, true)) === true) {
-            if ($store->id && $store->storeExists($store->id)) {
-                $res = $validateOnly ? $validateOnly : $store->update();
+        $fieldsValidationResult     = $store->validateFields(UNFRIENDLY_ERROR, true);
+        $langFieldsValidationResult = $store->validateFieldsLang(UNFRIENDLY_ERROR, true);
+
+        // If errors, log them and stop execution
+        if (true !== $fieldsValidationResult || true !== $langFieldsValidationResult) {
+            $errorGenericMessage = $this->trans('Store is invalid', [], 'Admin.Advparameters.Notification');
+            $this->errors[]      = $errorGenericMessage . ' (' . $store->name . ')';
+
+            $errorDetails = '';
+            if ($fieldsValidationResult !== true) {
+                $errorDetails .= $fieldsValidationResult;
             }
-            $store->force_id = (bool)$force_ids;
-            if (!$res) {
-                $res = $validateOnly ? $validateOnly : $store->add();
+            if ($langFieldsValidationResult !== true) {
+                $errorDetails .= $langFieldsValidationResult;
             }
+            $this->errors[] = $errorDetails;
 
-            if (!$res) {
-                $this->errors[] = Db::getInstance()->getMsgError().' '.sprintf(
+            return;
+        }
+
+        if ($validateOnly) {
+            return;
+        }
+
+        $res = false;
+        if ($store->storeExists($store->id)) {
+            $res = $store->update();
+        }
+
+        // If store doesn't exist or update failed
+        if (!$res) {
+            $store->force_id = (bool)$forceIds;
+            $res = $store->add();
+        }
+
+        // If nothing worked, log an error
+        if (!$res) {
+            $this->errors[] = Db::getInstance()->getMsgError() . ' '
+                . sprintf(
                     $this->trans('%1$s (ID: %2$s) cannot be saved', array(), 'Admin.Advparameters.Notification'),
                     $info['name'],
-                    (isset($info['id']) ? $info['id'] : 'null')
+                    ($store->id ? $store->id : 'null')
                 );
-            }
-        } else {
-            $this->errors[] = $this->trans('Store is invalid', array(), 'Admin.Advparameters.Notification').' ('.$store->name.')';
```

Comment:
@AlexEven, aren't we supposed to add `('.$store->name.')` in the translated string as a variable? This could be the occasion to update this.